### PR TITLE
add row index column for default sorting

### PR DESCRIPTION
**Pull request by mkangia**
https://dimagi-dev.atlassian.net/browse/ICDS-1237

##### SUMMARY
Mobile UCRs currently don't have explicit sorting supported. By default commcare sorts by the very first column, so this adds row index as the first column so that the order provided by HQ is maintained on the phone
Nit: picked name "row_index" instead of "index" to keep "index" being available as a name to be used in reports for any other purpose if needed. Since the column is getting added behind the scenes and is not visible on the mobile as well, user/app builder won't know about it. So picked a name that would not be used.

##### FEATURE FLAG
added new FF, `ADD_ROW_INDEX_TO_MOBILE_UCRS`

##### PRODUCT DESCRIPTION
No change on HQ. Report order would change on mobile to reflect what HQ provides and not the re-ordered list on mobile app based on whichever the first column is.

### CODE REVIEW

**Review by orangejenny — APPROVED** [review]
Looks good.

**Comment by orangejenny** [comment]
Test failures look like `test_report_module` needs to be updated to include the new column.

File: corehq/apps/app_manager/fixtures/mobile_ucr.py [hunk 1]
```diff
@@ -344,6 +346,8 @@ def _v2_fixtures(self, restore_user, report_configs):
     def report_config_to_fixture(self, report_config, restore_user):
         def _row_to_row_elem(deferred_fields, filter_options_by_field, row, index, is_total_row=False):
             row_elem = E.row(index=str(index), is_total_row=str(is_total_row))
+            if toggles.ADD_ROW_INDEX_TO_MOBILE_UCRS.enabled(restore_user.domain):
+                row_elem.append(E('row_index', str(index)))
```
Comment (esoergel) [inline corehq/apps/app_manager/fixtures/mobile_ucr.py]: nit: it looks like this `_row_to_row_elem` is identical for both v1 and v2, so you could probably pull it out.

**Review by esoergel — APPROVED** [review]
I don't recall - do you know whether there's some way to make a column appear in the report fixture but not when viewed on mobile?  Like a display property or something?  I wonder if this could be always added, but excluded from view.

**Comment by mkangia** [comment]
Not sure what you are suggesting here @esoergel, are you talking about making a hidden column?
if yes, that is what is happening here with width 0.

File: corehq/apps/app_manager/fixtures/mobile_ucr.py (hunk 1 above)
Comment (mkangia) [reply corehq/apps/app_manager/fixtures/mobile_ucr.py]: v1 is using E but v2 is using E.column for columns which makes them slightly different. The signature to both these methods is different as well, hence just passing them as an argument to us to a a function won't do.

**Comment by mkangia** [comment]
@esoergel merging this so that it can get rolled out soon. If there are more improvements, happy to take them in a follow up PR.

**Comment by esoergel** [comment]
@mkangia I was commenting on whether it would be feasible to do this change without a feature flag - should this be the default behavior for everyone?

**Comment by mkangia** [comment]
@esoergel took [clayton's suggestion](https://dimagi-dev.atlassian.net/browse/ICDS-1237?focusedCommentId=39805&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-39805) to not roll it out for everyone

**Pull request merged**